src/frontend/js/components/TeacherDashboardCourseList/index.tsx

```diff
@@ -29,28 +29,34 @@ const messages = defineMessages({
 interface TeacherDashboardCourseListProps {
   titleTranslated?: string;
   organizationId?: string;
+  loadMore: () => void;
```

Comment:
I might be a bit 🍺 but I don't understand why you had to add all those `useCourseProductUnion` to props to bring them up?

Reply:
How would you suggest that these information get given to `TeacherDashboardCourseList` from `TeacherDashboardOrganizationCourseLoader` and `TeacherDashboardCourseLoader` if i don't use props ?
calling `useCourseProductUnion` a second time here would not work.

Reply:
Why not using `useTeacherCoursesSearch` directly inside `TeacherDashboardCourseList`?

Reply:
I need information from it.
`TeacherDashboardCourseList` handle the render of the list.
The layout handle everything around, title, searchbar, counter.